cubeb-api/examples/common/mod.rs

```diff
@@ -1,29 +1,25 @@
 use cubeb::{Context, Result};
 use std::env;
 use std::ffi::CString;
-use std::io::{self, Write};
 
 pub fn init<T: Into<Vec<u8>>>(ctx_name: T) -> Result<Context> {
-    let backend = match env::var("CUBEB_BACKEND") {
-        Ok(s) => Some(s),
-        Err(_) => None,
-    };
+    // allow user to select backend via env var
+    let backend = env::var("CUBEB_BACKEND").ok();
 
+    // a little dance to convert to `Option<&CStr>`
+    let backend_c = backend.clone().map(|s| CString::new(s).unwrap());
+    let backend_c = backend_c.as_ref().map(|c| c.as_c_str());
+
+    // setup context
     let ctx_name = CString::new(ctx_name).unwrap();
-    let ctx = Context::init(Some(ctx_name.as_c_str()), None);
+    let ctx = Context::init(Some(ctx_name.as_c_str()), backend_c);
```

Comment:
Here's a small functional change, we actually try to select the provided backend.
FYI, on my Mac, I can only get `audiounit` to work, is `audiounit-rust` not provided by default?

Reply:
Not by default no, cubeb can be built without a rust toolchain. It's at https://github.com/mozilla/cubeb-coreaudio-rs/, https://github.com/mozilla/cubeb-coreaudio-rs/blob/trailblazer/build-audiounit-rust-in-cubeb.sh has the steps to build it, but really it's just cloning it in the `src/` directory of the cubeb dir, and building cubeb with `-DBUILD_RUST_LIBS="ON"`